mesonbuild/mlog.py

```diff
@@ -495,3 +495,17 @@ def stop_pager() -> None:
             pass
         log_pager.wait()
         log_pager = None
+
+
+def code_line(text: str, line: str, colno: int) -> str:
+    """Print a line with a caret pointing the colno
+
+    :param text: A message to display before the line
+    :param line: The line of code to be pointed to
+    :param colno: The column number to point at
+    :return: A formatted string of the text, line, and a caret
+    """
+    return f'''\
+        {text}
+        {line}
+        {' ' * colno}^'''
```

Comment:
This intersperses whitespace into the printed string, unlike the single-line version this is copied from.

Reply:
Is it really better to dedent this? Why not just use the pretty readable original string?

This is still broken, so maybe it's not all that readable using dedent. :)